Framework/Algorithms/src/FitPeaks.cpp

```diff
@@ -1079,16 +1093,66 @@ void FitPeaks::fitSpectrumPeaks(
 
     double expected_peak_pos = expected_peak_centers[peak_index];
 
-    bool foundAnyPeak = (lastGoodPeakParameters[fit_index].size() >
-                         static_cast<size_t>(std::count_if(
-                             lastGoodPeakParameters[fit_index].begin(),
-                             lastGoodPeakParameters[fit_index].end(),
-                             [&](auto const &val) { return val <= 1e-10; })));
-    if (foundAnyPeak) {
-      // set the peak parameters from last good fit to that peak
+    // Determine how to set the starting value parameters
+
+    // Determine whether to set starting parameter from fitted value
+    // of same peak but different spectrum
+    bool samePeakCrossSpectrum =
+        (lastGoodPeakParameters[fit_index].size() >
+         static_cast<size_t>(
+             std::count_if(lastGoodPeakParameters[fit_index].begin(),
+                           lastGoodPeakParameters[fit_index].end(),
+                           [&](auto const &val) { return val <= 1e-10; })));
+    // Check whether current spectrum's pixel (detector ID) is close to its
+    // previous spectrum's pixel (detector ID).
+    try {
+      if (wi > 0 && samePeakCrossSpectrum) {
+        // First spectrum or discontinuous detector ID: do not start from same
+        // peak of last spectrum
+        std::shared_ptr<const Geometry::Detector> pdetector =
+            std::dynamic_pointer_cast<const Geometry::Detector>(
+                m_inputMatrixWS->getDetector(wi - 1));
+        std::shared_ptr<const Geometry::Detector> cdetector =
+            std::dynamic_pointer_cast<const Geometry::Detector>(
+                m_inputMatrixWS->getDetector(wi));
+
+        // If they do have detector ID
+        if (pdetector && cdetector) {
+          auto prev_id = pdetector->getID();
+          auto curr_id = cdetector->getID();
+          if (prev_id + 1 != curr_id)
+            samePeakCrossSpectrum = false;
+        } else {
+          samePeakCrossSpectrum = false;
+        }
+
+      } else {
+        // first spectrum in the workspace: no peak's fitting result to copy
+        // from
+        samePeakCrossSpectrum = false;
+      }
+    } catch (const std::runtime_error &) {
+      // workspace does not have detector ID set: there is no guarantee that the
+      // adjacent spectra can have similar peak profiles
+      samePeakCrossSpectrum = false;
+    }
+
+    // Set starting values of the peak function
```

Comment:
So setting the centre and SetMatrixWorkspace (currently on L1162) needs to happen up here (before get parameters cross or same spectrum) - even if we find any peaks so as to get A and B if it is a BackToBackExponential and the Parameters.xml file contains alpha0 etc. (it won't effect any parameters in another peak shape). In addition we then don't want to overwrite A and B later, so instead of copying all the parameters from the appropriate peak (cross or same spectrum) we instead only want to set FWHM and height (although as discussed for the BackToBackExponential if the coeficients in the  PArameters.xml file are not good the fwhm will not be set well - we should maybe add to the docs that we don't advise fitting with BackToBackExponential without Parameters.xml file present). But I am happy to address these concerns this in a separate PR.